datapackage_pipelines_measure/processors/add_twitter_resource.py

```diff
@@ -0,0 +1,285 @@
+# Followers
+# This is requested directly from the twitter api
+
+followers_count = None
+if entity_type == 'account':
+    user = _get_user_account_from_twitter_api(entity)
+    followers_count = user.followers_count
+
+
+# Mentions & Interactions
+# These are requested for specified (and limited) timeframes from the twitter
+# api, then added to the previous result to get a total since we started
+# collecting data for the entity.
+
+# Get last run entity values
+filter_object = {
+    'entity': entity,
+    'entity_type': entity_type,
+    'source': 'twitter',
+    'project_id': project_id
+}
+datastore = get_datastore()
+entity_last_run = datastore.get_latest_from_table(filter_object,
+                                                  DATASTORE_TABLE)
+mentions = 0
+interactions = 0
+if entity_last_run:
+    latest_mentions = entity_last_run.get('mentions', 0)
+    latest_interactions = entity_last_run.get('interactions', 0)
+    if entity_last_run['timestamp'].date() == datetime.date.today():
+        # Last run today
+        log.debug('Last run for entity, "{}". was today.'.format(entity))
+        # Don't collect from twitter again, just reuse entity_last_run's data
+        mentions = latest_mentions
+        interactions = latest_interactions
+    elif entity_last_run['timestamp'].date() < datetime.date.today():
+        # Last run before today, get tweets from the last run date to
+        # yesterday.
+        log.debug('Last run for entity, "{}", was before today.'
+                  .format(entity))
+        end_date = datetime.date.today()  # end_date is exclusive (so yesteray)
+        start_date = entity_last_run['timestamp'].date()
+        mentions = _get_mentions_for_entity_from_source(
+            entity,
+            start_date.strftime(TWITTER_API_DATE_RANGE_FORMAT),
+            end_date.strftime(TWITTER_API_DATE_RANGE_FORMAT)
+        )
+        mentions = mentions + latest_mentions
+        interactions = _get_interactions_for_entity_from_source(
+            entity,
+            start_date.strftime(TWITTER_API_DATE_RANGE_FORMAT),
+            end_date.strftime(TWITTER_API_DATE_RANGE_FORMAT)
+        )
+        interactions = interactions + latest_interactions
+else:
+    # No last run for this entity.
+    log.debug('No last run for entity "{}"'.format(entity))
+    # Collect twitter data for entity for the last few days
+    end_date = datetime.date.today()  # end_date is exclusive (so yesterday)
+    start_date = (end_date - datetime.timedelta(
+        days=TWITTER_API_SEARCH_INDEX_LIMIT_IN_DAYS))
+    mentions = _get_mentions_for_entity_from_source(
+        entity,
+        start_date.strftime(TWITTER_API_DATE_RANGE_FORMAT),
+        end_date.strftime(TWITTER_API_DATE_RANGE_FORMAT)
+    )
+    interactions = _get_interactions_for_entity_from_source(
+        entity,
+        start_date.strftime(TWITTER_API_DATE_RANGE_FORMAT),
+        end_date.strftime(TWITTER_API_DATE_RANGE_FORMAT)
+    )
+
+resource_content = {
```

Comment:
no timestamp in record? 
I think that we should collect the raw data here and do aggregations later.